### PR TITLE
Deprecate and cleanup `OC.L10N` in favor of `@nextcloud/l10n` 

### PR DESCRIPTION
**Pull request by susnux**
## Summary
This PR consists of three parts:
1. Cleanup long standing deprecation from `OC`: `OC.addTranslations` was deprecated with NC17 and is not used anymore by any app within the github nextcloud organization
2. Deprecate `OC.getLocale` and `OC.getLanguage` as they are provided by `@nextcloud/l10n` and drop duplicated code
3. Deprecate and replace the translation functions implementation with the `@nextcloud/l10n`.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated
  - See PR: https://github.com/nextcloud/documentation/pull/9575 (upgrade guide)

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]
:broom:

**Review by Pytal — APPROVED** [review]

File: core/src/OC/l10n.js [hunk 1]
```diff
@@ -28,79 +27,56 @@
...
-		registerAppTranslations(appName, bundle, this._getPlural)
-	},
+	register,
 
 	/**
 	 * @private
+	 * @deprecated 26.0.0 use `_unregister` from https://www.npmjs.com/package/@nextcloud/l10n
 	 */
-	_unregister: unregisterAppTranslations,
+	_unregister,
```
Comment (skjnldsv) [inline core/src/OC/l10n.js:73]: is it a private method? Then it should not be exposed or it should be made public?

**Review by skjnldsv — APPROVED** [review]

File: core/src/OC/l10n.js (hunk 1 above)
Comment (susnux) [reply core/src/OC/l10n.js:73]: This is currently only used in CI tests, but I do not see a reason why this function should not be on the public API.
Maybe we should change this on the `@nextcloud/l10n` package.

But for this PR I wanted to keep the changes small and do not introduce a breaking API change.

**Comment by susnux** [comment]
Rebased onto current master

**Comment by susnux** [comment]
Blocked by nextcloud/nextcloud-l10n#571 (and we need to wait for a new release)
There was an issue introduced in the standalone implementation which caused the function to synchronously load translations instead of asynchronously, this is the reason the test fail here.

**Comment by susnux** [comment]
The drone CI failure on the mysql DB tests seem unrelated.

**Pull request merged**